From ccff8a426babb9a34e64b0024ce4d651c1ca25e5 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 17 Jan 2014 12:51:24 -0500
Subject: [PATCH] seat: always add text displays when opening seat

At the moment we add pixel displays if we can, or
text displays if we can't add pixel displays.

We need to always add text displays, otherwise, the
text splash won't work when explicitly configured by
the user.
---
 src/libply-splash-core/ply-seat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libply-splash-core/ply-seat.c b/src/libply-splash-core/ply-seat.c
index 0900346..d9c7a2b 100644
--- a/src/libply-splash-core/ply-seat.c
+++ b/src/libply-splash-core/ply-seat.c
@@ -110,62 +110,62 @@ bool
 ply_seat_open (ply_seat_t          *seat,
                ply_renderer_type_t  renderer_type,
                const char          *device)
 {
   if (renderer_type != PLY_RENDERER_TYPE_NONE)
     {
       ply_renderer_t *renderer;
 
       renderer = ply_renderer_new (renderer_type, device, seat->terminal);
 
       if (!ply_renderer_open (renderer) && renderer_type != PLY_RENDERER_TYPE_AUTO)
         {
           ply_trace ("could not open renderer for %s", device);
           ply_renderer_free (renderer);
           return false;
         }
 
       seat->renderer = renderer;
       seat->renderer_active = true;
     }
 
   if (seat->renderer != NULL)
     {
       seat->keyboard = ply_keyboard_new_for_renderer (seat->renderer);
       add_pixel_displays (seat);
 
     }
   else
     {
       seat->keyboard = ply_keyboard_new_for_terminal (seat->terminal);
-      add_text_displays (seat);
     }
+  add_text_displays (seat);
 
   ply_keyboard_watch_for_input (seat->keyboard);
   seat->keyboard_active = true;
 
   return true;
 }
 
 bool
 ply_seat_is_open (ply_seat_t *seat)
 {
   return ply_list_get_length (seat->pixel_displays) > 0 ||
          ply_list_get_length (seat->text_displays) > 0;
 }
 
 void
 ply_seat_deactivate_keyboard (ply_seat_t *seat)
 {
   if (!seat->keyboard_active)
     return;
 
   seat->keyboard_active = false;
 
   if (seat->keyboard == NULL)
     return;
 
   ply_trace ("deactivating keybord");
   ply_keyboard_stop_watching_for_input (seat->keyboard);
 }
 
 void
-- 
1.8.3.1

